paddleseg/models/enet.py

```diff
@@ -0,0 +1,759 @@
+                self.stride = stride
+        if isinstance(padding, int):
+            self.padding = (padding, padding)
+        else:
+            self.padding = padding
+        self.align_corners = align_corners
+
+    def forward(self, input, indices, output_size=None):
+
+        # pdb.set_trace()
+        if output_size is None:
+            n, c, h, w = input.shape
+            out_h = (h - 1) * self.stride[0] - 2 * \
+                self.padding[0] + self.kernel_size[0]
+            out_w = (w - 1) * self.stride[1] - 2 * \
+                self.padding[1] + self.kernel_size[1]
+            output_size = (n, c, out_h, out_w)
+        out = F.interpolate(input,
+                            output_size[2:],
+                            mode='bilinear',
+                            align_corners=self.align_corners)
+        # # else:
+        # #     if len(output_size) == len(self.kernel_size) + 2:
+        # #         output_size = output_size[2:]
+        # t = str(input.dtype).lower().strip().split(".")[-1]
+        # # print(t)
+        # t = TYPE_MAPPER[t]
+        # out = paddle.zeros(output_size, dtype=t)
+        # flatten_out = paddle.flatten(out)
+        # pdb.set_trace()
+        # for i in range(indices.shape[0]):
+        #     for j in range(indices.shape[1]):
+        #         for k in range(indices.shape[2]):
+        #             for m in range(indices.shape[3]):
+        #                 indices[i, j, k, m] = (out.shape[1] * out.shape[2] * out.shape[3]) * i + \
+        #                                       (out.shape[2] * out.shape[3]) * j + indices[i, j, k, m]
+        # flatten_indices = paddle.flatten(indices)
+        # flatten_input = paddle.flatten(input)
+        # for i in range(flatten_indices.shape[0]):
+        #     flatten_out[flatten_indices[i].tolist()] = flatten_input[i].tolist()
+        # out = paddle.reshape(flatten_out, out.shape)
+        return out
+
+
+class InitialBlock(nn.Layer):
+    """The initial block is composed of two branches:
+    1. a main branch which performs a regular convolution with stride 2;
+    2. an extension branch which performs max-pooling.
+    Doing both operations in parallel and concatenating their results
+    allows for efficient downsampling and expansion. The main branch
+    outputs 13 feature maps while the extension branch outputs 3, for a
+    total of 16 feature maps after concatenation.
+    Keyword arguments:
+    - in_channels (int): the number of input channels.
+    - out_channels (int): the number output channels.
+    - kernel_size (int, optional): the kernel size of the filters used in
+    the convolution layer. Default: 3.
+    - padding (int, optional): zero-padding added to both sides of the
+    input. Default: 0.
+    - bias (bool, optional): Adds a learnable bias to the output if
+    ``True``. Default: False.
+    - relu (bool, optional): When ``True`` ReLU is used as the activation
+    function; otherwise, PReLU is used. Default: True.
```

Comment:
the format is not correct, plz refer to other model implementations

Reply:
Done.